src/goto-instrument/contracts/contracts.cpp

```diff
@@ -705,7 +704,7 @@ void code_contractst::instrument_assign_statement(
     instruction_iterator->source_location));
   alias_assertion.instructions.back().source_location.set_comment(
     "Check that " + from_expr(ns, lhs.id(), lhs) + " is assignable");
-  int lines_to_iterate = alias_assertion.instructions.size();
+  size_t lines_to_iterate = alias_assertion.instructions.size();
   program.insert_before_swap(instruction_iterator, alias_assertion);
   std::advance(instruction_iterator, lines_to_iterate);
```

Comment:
If you rebase your branch on to the current `develop`, you wouldn't need to keep track of `lines_to_iterate` any more. I added a helper function for this, and you would be able to do:

```c
insert_before_swap_and_advance(program, instruction_iterator, alias_assertion);
```

(same for all other occurrences of `insert_before_swap` followed by `std::advance` pattern in this file)

Reply:
Great! Implemented as suggested!